src/rocdata.jl

```diff
@@ -22,90 +23,91 @@ function _vector2labels(labels, truelabel)
     end
 end
 
-function _preparedrocdata(scores, labels, reverseordered::Bool)
+function _preparedrocdata(scores, labels, distancescored)
 	if length(scores) == length(labels)
-		_create_preparedrocdata(scores,labels,reverseordered)
+		_create_preparedrocdata(scores,labels,distancescored)
 	else
 		error("scores and labels should have the same length")
 	end
 end
 
 struct ROCData{T<:Real}
-	scores::Vector{T}
-	labels::BitVector
+	scores::AbstractVector{T}
+	labels::AbstractVector{Bool}
 	P::Int
 	n::Int
 	N::Int
 	ni::UnitRange{Int}
-	TP::Vector{Int}
-	TN::Vector{Int}
-	FP::Vector{Int}
-	FN::Vector{Int}
-	FPR::Vector{Float64}
-	TPR::Vector{Float64}
+	TP::AbstractVector{Int}
+	TN::AbstractVector{Int}
+	FP::AbstractVector{Int}
+	FN::AbstractVector{Int}
+	FPR::AbstractVector{Float64}
+	TPR::AbstractVector{Float64}
 end
 
 function roc(_preparedrocdata::_PreparedROCData)
 	P = sum(_preparedrocdata.labels)
-	n = length(_preparedrocdata.labels)
+	n = length(_preparedrocdata.labels)+1
 	N = n - P
-	ni = 1:(n-1)
-	TP = Array{Int}(ni.stop)
-	TN = Array{Int}(ni.stop)
-	FP = Array{Int}(ni.stop)
-	FN = Array{Int}(ni.stop)
-	FPR = Array{Float64}(ni.stop)
-	TPR = Array{Float64}(ni.stop)
+	ni = 2:n
```

Comment:
Since then you do `point = i - 1` this is the same as `1:(n-1)`. But since `n` is now 1 point greater than the length, my previous `1:(n-1)` with `n` equal to length should have been `1:n` (I was missing the last point).